plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/hms/AbstractMetastoreTableOperations.java

```diff
@@ -124,10 +123,11 @@ protected final void commitNewTable(TableMetadata metadata)
         try {
             metastore.createTable(table, privileges);
         }
-        catch (SchemaNotFoundException | TableAlreadyExistsException e) {
-            // clean up metadata files corresponding to the current transaction
+        catch (Exception e) {
+            // clean up metadata file corresponding to the current transaction
             fileIo.deleteFile(newMetadataLocation);
-            throw e;
+            // wrap exception in CleanableFailure to ensure that manifest list Avro files are also cleaned up
```

Comment:
Wondering should we delete the table folder as well? If the table does not exist it creates ir, and if user for not retry, folder will remain there.

Reply:
Not neccecery. The test actually ensures the table can be created in the same location afterwards